### PR TITLE
Savya/case manager

### PR DESCRIPTION
**Pull request by msavya**
# Pull Request

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Related Issues (put task name here from notion)

Case Manager Request Form

### Code Review Preparation
- [ ] I have performed a self-review and reviewed the changed files
- [ ] My code follows the project's style guidelines and passes linting
- [ ] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [ ] I have added relevant tests and they pass locally
- [ ] I have tested the changes manually, including edge cases

### Code Quality & Security
- [ ] I have formatted my code and removed debugging/unused code
- [ ] No TypeScript errors or linting issues
- [ ] No sensitive information committed (API keys, passwords, etc.)
- [ ] User inputs are validated and sanitized

### Dependencies & Deployment
- [ ] Dependencies are documented and package-lock.json is updated if needed
- [ ] Changes work with Firebase emulator and environment variables are documentedRetry

## Screenshots (If it is a front end feature screenshot is required)

Screenshot 2026-02-17 at 4 34 21 PM
Screenshot 2026-02-17 at 4 35 16 PM
Screenshot 2026-02-17 at 4 35 26 PM
Screenshot 2026-02-17 at 4 35 42 PM

## Release Notes

* **New Features**
  * Added multi-step client request form with four stages: client information collection, item requests by category (bedding, tables, kitchen, other), review, and confirmation.
  * Implemented client request submission and data persistence.
  * Added case manager access restrictions to the form.
  * Enhanced step indicators with descriptive labels.

* **Bug Fixes & Improvements**
  * Added form validation with error messaging.
  * Improved form inputs with disabled state and enhanced value handling.
  * Added phone number formatting and date conversion support.

### CODE REVIEW

File: app/client-request-form/steps/Step1ClientInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+                                            clearError("zipCode");
+                                        }}
+                                    />
+                                    {errors.zipCode && (
+                                        <p className="text-red-500 text-sm mt-1">
+                                            {errors.zipCode}
+                                        </p>
+                                    )}
+                                </div>
+                            </div>
```
[the remark this answers is not in the document]
Comment (msavya) [reply app/client-request-form/steps/Step1ClientInfo.tsx]: Not sure why the street address needs to be made optional if it's a required field

File: app/client-request-form/steps/Step3Review.tsx [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+                { name: "Towel set", quantity: formState.other.towelset },
+                { name: "Lamp", quantity: formState.other.lamp },
+            ].filter((item) => item.quantity > 0);
+
+            const uid = auth.currentUser?.uid;
+            if (!uid) {
+                alert("You must be logged in to submit a request.");
+                return;
+            }
+            const request: ClientRequest = {
```
Comment (joelchem) [inline app/client-request-form/steps/Step3Review.tsx]: Could you add a "date" field to the client requests type and also populate it here

**Review by joelchem — APPROVED** [review]
